src/app/zap-templates/templates/app/cluster-objects.zapt

```diff
@@ -63,7 +69,55 @@ namespace {{name}} {
 } // namespace {{name}}
 {{/zcl_structs}}
 
-} // namespace {{asUpperCamelCase name}}
+namespace Commands {
+{{#zcl_commands}}
+namespace {{name}}CommandParams {
+constexpr CommandId kCommandId = {{asHex code 2}};
+
+enum FieldId {
+    {{#zcl_command_arguments}}
+    k{{asUpperCamelCase label}}FieldId = {{index}},
+    {{/zcl_command_arguments}}
+};
+
+struct Type {
+public:
+    {{#zcl_command_arguments}}
+    {{#if isArray}}
+    DataModel::List<{{#if_is_enum type}}{{asChipEnumType type}}{{else}}{{asChipZapType type}}{{/if_is_enum}}> {{asLowerCamelCase label}};
+    {{else}}
+    {{#if_is_enum type}}{{asChipEnumType type}}{{else}}{{asChipZapType type}}{{/if_is_enum}} {{asLowerCamelCase label}};
+    {{/if}}
+    {{/zcl_command_arguments}}
+
+    CHIP_ERROR Encode(TLV::TLVWriter &writer, uint64_t tag) const;
+    CHIP_ERROR Decode(TLV::TLVReader &reader);
+};
+
+{{#if command_contains_array}}
+struct DecodableType {
+public:
+    {{#zcl_command_arguments checkForDoubleNestedArray="true"}}
+    {{#if isArray}}
+    DataModel::DecodableList<{{#if command_arg_contains_nested_array}}{{type}}::DecodableType{{else}}{{#if_is_enum type}}{{asChipEnumType type}}{{else}}{{asChipZapType type}}{{/if_is_enum}}{{/if}}> {{asLowerCamelCase label}};
+    {{else}}
+    {{#if command_arg_contains_nested_array}}{{type}}::DecodableType{{else}}{{#if_is_enum type}}{{asChipEnumType type}}{{else}}{{asChipZapType type}}{{/if_is_enum}}{{/if}} {{asLowerCamelCase label}};
```

Comment:
Can we use the machinery from #10175 here and other places?  That is, just use `zapTypeToDecodableClusterObjectType` and if it does not do the right thing for enums fix it to do that?  I am pretty sure that after #10175 we don't need the `command_arg_contains_nested_array` check: we just use `DecodableType` in the struct case and the right thing will happen, which is what `zapTypeToDecodableClusterObjectType` ends up doing.